src/Lexer.test.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) ZALORA
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+import Lexer from './Lexer';
+import { dot, int, ident, lbracket, rbracket, eof } from './Tokens';
+
+describe('Lexer', () => {
+  it('tests with empty input', () => {
+    const input = '';
+    const testCase = { type: eof, literal: '' };
+    const l = new Lexer(input);
+
+    const token = l.nextToken();
+    expect(token).toMatchObject(testCase);
+  });
+
+  it('tests with one key', () => {
+    const input = '.data';
+    const testCases = [
```

Comment:
In the iOS implementation, lexer output always ends with eof, and the tests test for that https://github.com/zalora/pluto/blob/master/JSL/JSLTests/LexerTests.swift. Please make this the same.

Reply:
sure 👍🏻